src/commands.cpp

```diff
@@ -4834,23 +4846,70 @@ bool CommandSetPH::procresult(Result r)
         return true;
     }
 
-    handle ph = client->json.gethandle();
 
-    if (ISUNDEF(ph))
+    if (mWritable) // aparently, depending on 'w', the response can be [{"ph":"XXXXXXXX","w":"YYYYYYYYYYYYYYYYYYYYYY"}] or simply [XXXXXXXX]
     {
-        client->app->exportnode_result(API_EINTERNAL);
-        return true;
-    }
+        handle ph = UNDEF;
+        std::string authKey;
 
-    Node *n = client->nodebyhandle(h);
-    if (n)
+        for (;;)
+        {
+            switch (client->json.getnameid())
+            {
+            case 'w':
+                client->json.storeobject(&authKey);
+                break;
+
+            case MAKENAMEID2('p', 'h'):
+                ph = client->json.gethandle();
+                break;
+
+            case EOO:
+            {
+                if (ISUNDEF(ph))
```

Comment:
Shouldn't we check the auth key was received?
```suggestion
                if (ISUNDEF(ph) || authKey.empty())
```